Example/Sources/ConversationViewController.swift

```diff
@@ -0,0 +1,106 @@
+                   "Quam erat vehicula metus, et condimentum ante tellus augue."
+        
+        let msg5 = "Lorem ipsum dolor sit amet, consectetur adipiscing elit." +
+                   "Pellentesque venenatis, ante et hendrerit rutrum" +
+                   "Quam erat vehicula metus, et condimentum ante tellus augue."
+
+        messages.append(MockMessage(text: msg2, sender: sender2, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg4, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg5, sender: sender3, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg1, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg3, sender: sender1, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg3, sender: sender1, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg2, sender: sender2, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg2, sender: sender2, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg1, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg3, sender: sender1, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg2, sender: sender2, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg4, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg5, sender: sender3, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg4, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg5, sender: sender3, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg4, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg5, sender: sender3, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg1, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg1, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg3, sender: sender1, id: NSUUID().uuidString))
+    }
+    
+    func currentSender() -> Sender {
+        return Sender(id: "123", displayName: "Steven")
+    }
+    
+    func numberOfMessages(in collectionView: UICollectionView) -> Int {
+        return messages.count
+    }
+    
+    func messageForItem(at indexPath: IndexPath, in collectionView: UICollectionView) -> MessageType {
+        return messages[indexPath.section]
+    }
+    
+    func avatarForMessage(_ message: MessageType, at indexPath: IndexPath, in collectionView: UICollectionView) -> Avatar {
+        let image = isFromCurrentSender(message: message) ? #imageLiteral(resourceName: "Steve-Jobs") : #imageLiteral(resourceName: "Tim-Cook")
+        return Avatar(placeholderImage: image)
```

Comment:
@SD10 I agree that it's difficult to build easy customized API. Thanks for the optional type in swift, we can simple handle who don't want Avatars by putting a question mark.

I like the idea using protocol for some models. Swift is basically protocol oriented programming language. It's amazing that protocol allows inheritance and extension to make customize and default values easy. But my concern is that we are build a framework for all potential use. From a user's perspective, I don't want to build all the models which are declared as protocol in the library. I just want to it easy to use.